spec/core/QueueRunnerSpec.js

```diff
@@ -234,6 +234,52 @@ describe("QueueRunner", function() {
       queueRunner.execute();
       expect(doneReturn).toBe(null);
     });
+
+    it("supports promises (and thenables) returned by queued functions", function(done) {
+      var queueableFn1 = { fn: function() {
+          return Promise.resolve();
+        } },
+        queueableFn2 = { fn: jasmine.createSpy('fn2') },
+        completeCallback = jasmine.createSpy('completeCallback'),
+        queueRunner = new jasmineUnderTest.QueueRunner({
+          queueableFns: [queueableFn1, queueableFn2],
+          onComplete: completeCallback
+        });
+
+      queueRunner.execute();
+
+      expect(completeCallback).not.toHaveBeenCalled();
+      expect(queueableFn2.fn).not.toHaveBeenCalled();
+
+      setImmediate(function() {
```

Comment:
Similarly, `setImmediate` isn't available in all browsers that Jasmine supports and as such can't be used in Jasmine's tests.